src/hotspot/share/gc/g1/heapRegion.cpp

```diff
@@ -459,153 +459,71 @@ void HeapRegion::print_on(outputStream* st) const {
   st->print_cr("");
 }
 
-class G1VerificationClosure : public BasicOopIterateClosure {
-protected:
-  G1CollectedHeap* _g1h;
+class VerifyRemSetClosure : public G1VerificationClosure {
   G1CardTable *_ct;
-  oop _containing_obj;
-  bool _failures;
-  int _n_failures;
-  VerifyOption _vo;
-
-public:
-
-  G1VerificationClosure(G1CollectedHeap* g1h, VerifyOption vo) :
-    _g1h(g1h), _ct(g1h->card_table()),
-    _containing_obj(NULL), _failures(false), _n_failures(0), _vo(vo) {
-  }
-
-  void set_containing_obj(oop obj) {
-    _containing_obj = obj;
-  }
-
-  bool failures() { return _failures; }
-  int n_failures() { return _n_failures; }
-
-  void print_object(outputStream* out, oop obj) {
-#ifdef PRODUCT
-    Klass* k = obj->klass();
-    const char* class_name = k->external_name();
-    out->print_cr("class name %s", class_name);
-#else // PRODUCT
-    obj->print_on(out);
-#endif // PRODUCT
-  }
-};
-
-class VerifyLiveClosure : public G1VerificationClosure {
 
   template <class T>
   void do_oop_work(T* p) {
-    assert(_containing_obj != NULL, "Precondition");
-    assert(!_g1h->is_obj_dead_cond(_containing_obj, _vo),
-      "Precondition");
+    assert(_containing_obj != nullptr, "Precondition");
+    assert(!_g1h->is_obj_dead_cond(_containing_obj, _vo), "Precondition");
 
     T heap_oop = RawAccess<>::oop_load(p);
     if (CompressedOops::is_null(heap_oop)) {
       return;
     }
 
     oop obj = CompressedOops::decode_raw_not_null(heap_oop);
-    bool is_in_heap = _g1h->is_in(obj);
-    if (!is_in_heap || _g1h->is_obj_dead_cond(obj, _vo)) {
-      MutexLocker x(ParGCRareEvent_lock, Mutex::_no_safepoint_check_flag);
 
-      Log(gc, verify) log;
-      if (!_failures) {
-        log.error("----------");
-      }
-      ResourceMark rm;
+    HeapRegion* from = _g1h->heap_region_containing_or_null(p);
+    HeapRegion* to = _g1h->heap_region_containing_or_null(obj);
+    if (from != nullptr && to != nullptr &&
+      from != to &&
+      !to->is_pinned() &&
+      to->rem_set()->is_complete()) {
```

Comment:
Mis-indented multi-line tests.  These are part of the if-condition, so should be lined up inside the `()`.
With the indentation being used it's not so easy to spot where the condition stops and the body ends.
Although I see it was like that before the change too, except this change adds a blank line separator.
I don't think the blank line is enough (easy to miss it's importance, esp. in a review diff), and these
lines should be indented further.